test/lib/ufe/testAttributeBlock.py

```diff
@@ -261,5 +261,91 @@ def testAttributeBlocking3dCommonApi(self):
+        in stronger layer(s) when using UsdTransform3dMatrixOp.
+        '''
+
+        # create new stage
+        cmds.file(new=True, force=True)
+
+        # Open usdCylinder.ma scene in testSamples
+        mayaUtils.openCylinderScene()
+
+        # get the stage
+        proxyShapes = cmds.ls(type="mayaUsdProxyShapeBase", long=True)
+        proxyShapePath = proxyShapes[0]
+        stage = mayaUsd.lib.GetPrim(proxyShapePath).GetStage()
+
+        # cylinder prim
+        cylinderPrim = stage.GetPrimAtPath('/pCylinder1')
+        self.assertIsNotNone(cylinderPrim)
+
+         # create 3 sub-layers ( LayerA, LayerB, LayerC ) and set the edit target to LayerB.
+        rootLayer = stage.GetRootLayer()
+        subLayerC = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerC")[0]
+        subLayerB = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerB")[0]
+        subLayerA = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerA")[0]
+
+        # check to see the sublayers added
+        addedLayers = [subLayerA, subLayerB, subLayerC]
+        self.assertEqual(rootLayer.subLayerPaths, addedLayers)
+
+        # set the edit target to LayerB
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerB)
+
+        # create a xformable and give it a matrix4d xformOp:transform
+        # This will match the UsdTransform3dMatrixOp, but not the Maya API.
+        cylinderXformable = UsdGeom.Xformable(cylinderPrim)
+        transformOp = cylinderXformable.AddTransformOp()
+        xform = Gf.Matrix4d(1.0).SetTranslate(Gf.Vec3d(10, 20, 30))
+        transformOp.Set(xform)
+        self.assertTrue(cylinderPrim.GetPrim().HasAttribute("xformOp:transform"))
+
+        # Now that we have set up the transform stack to be 3dMatrixOp compatible,
+        # create the Transform3d interface.
+        cylinderPath = ufe.Path([
+                     mayaUtils.createUfePathSegment("|mayaUsdTransform|shape"), 
+                     usdUtils.createUfePathSegment("/pCylinder1")])
+        cylinderItem = ufe.Hierarchy.createItem(cylinderPath)
+
+        cylinderT3d = ufe.Transform3d.transform3d(cylinderItem)
+
+        # select the cylinderItem
+        sn = ufe.GlobalSelection.get()
+        sn.clear()
+        sn.append(cylinderItem)
+
+        # move
+        cmds.move(65, 20, 10)
+        self.assertEqual(cylinderXformable.GetXformOpOrderAttr().Get(),
+                    Vt.TokenArray(('xformOp:transform', )))
+
+        # validate the Matrix4d entires
+        actual = cylinderXformable.GetLocalTransformation()
+        expected  = Gf.Matrix4d( 1, 0, 0, 0,
+                                 0, 1, 0, 0,
+                                 0, 0, 1, 0,
+                                 65, 20, 10, 1)
+
+        self.assertTrue(Gf.IsClose(actual, expected, 1e-5))
```

Comment:
Line 321 to 332: Set the translate to a new value `[65, 20, 10]` and check the 4x4 affine matrix with the newly translate values.